src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentContext.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.Workspaces.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal record DocumentContext
```

Comment:
This is the magical type that I'm expecting we can start to use in order to simplify feature endpoint flows. I considered adding helper methods here for acquiring syntax nodes from the syntax tree etc. but figured that's probably better done as extension methods to the RazorSyntaxTree API OR one day maybe the syntax tree will have that first class 😆